Add script to open Claude and submit prompt

nimobeeren opened the pull request:
## Description

This script opens Claude in Chrome browser and submits the prompt provided as argument.

## Type of change

<!-- Please delete options that are not relevant. -->

New script command

## Screenshot

https://github.com/user-attachments/assets/49f21e5f-5574-44c5-a559-4719ca94db28


## Dependencies / Requirements

<!-- If it's a new script command that requires some additional steps to make it work, please describe it here. E.g. requiring installation of another command line tool or requirement to specify username / API token / etc. -->

This script requires `chrome-cli` installed: https://github.com/prasmussen/chrome-cli

Install via homebrew: `brew install chrome-cli`

## Checklist

- [x] I have read [Contribution Guidelines](https://github.com/raycast/script-commands/blob/master/CONTRIBUTING.md)

unnamedd commented on commands/apps/claude/claude.js:
```diff
@@ -0,0 +1,120 @@
+
+// Optional parameters:
+// @raycast.icon 🧠
+// @raycast.argument1 { "type": "text", "placeholder": "prompt"}
```
The placeholders need to be camel case

```suggestion
// @raycast.argument1 { "type": "text", "placeholder": "Prompt"}
```

unnamedd commented on commands/apps/claude/claude.js:
```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env node
+
+// Dependency: This script requires `chrome-cli` installed: https://github.com/prasmussen/chrome-cli
+// Install via homebrew: `brew install chrome-cli`
```
Would be better if you note here that `node` is another requirement, since `node` isn't something pre-installed on macOS.

nimobeeren replied:
done

unnamedd commented on commands/apps/claude/claude.js:
```diff
@@ -0,0 +1,120 @@
+
+// Required parameters:
+// @raycast.schemaVersion 1
+// @raycast.title Prompt Claude
```
Just an idea, but what do you think about
```suggestion
// @raycast.title Ask Claude
```
?

unnamedd commented on commands/apps/claude/claude.js:
```diff
@@ -0,0 +1,120 @@
+  execSync("which chrome-cli");
+} catch {
+  console.error(
+    "chrome-cli is required to run this script (https://github.com/prasmussen/chrome-cli)"
```
This means your Script Command will only work on Chrome, maybe you should open the usage of your Script Command to users of other browsers, even though, many of them uses Chromium as engine, sounds to me that `chrome-cli` will work only for Chrome.

Is something that important to be able to search for an open Claude tab in the browser to reuse it with a new prompt?

nimobeeren replied:
`chrome-cli` indeed works with other Chromium-based browsers through aliases such as `brave-cli` (see [the `chrome-cli` readme](https://github.com/prasmussen/chrome-cli/tree/master?tab=readme-ov-file#other-browsers)). However, I'm not sure how to detect which browser the user would prefer. The most straight-forward approach I see is letting the user modify this script if they want to use another browser.

I could add a variable at the top of the script to make it easier to change the `chrome-cli` executable. What do you think is best?

unnamedd replied:
Nice @nimobeeren!

Reading it now, it looks much simpler, which led me to another question (and this is not a blocker, but something for you to consider): would it not be better if you convert from Node to a Shell script? Since you are just checking whether `chrome-cli` is installed or not, perhaps you can do all that process in Shell, avoiding even the need for a Node installation.

unnamedd replied:
> I could add a variable at the top of the script to make it easier to change the `chrome-cli` executable. What do you think is best?

Maybe in this case, you could make the extension a template (just adding `.template` in the file name and tell the user to uncomment the line related to its browser: brave, chrome and so on. What do you think?

unnamedd reviewed (changes requested):
Hi @nimobeeren, thanks for you first contribution to our repository,

I've made some small comments on your PR, give a check and re-request a review when you finish to address them.

nimobeeren commented:
@unnamedd thanks for your review! I've implemented or replied to your comments.

I've also changed behavior to open in a new tab instead of new window, and added a note to tell the user to enable JavaScript from Apple Events. Could you give it another look?

unnamedd reviewed (approved):
Hey @nimobeeren,

thanks for you contribution! Even though there are something we could improve here, it looks good to me! 

If you prefer to make improvements later, will be a pleasure to review them later!

unnamedd commented:
@nimobeeren I found this link earlier today ([Claude’s API now supports CORS requests, enabling client-side applications](https://simonwillison.net/2024/Aug/23/anthropic-dangerous-direct-browser-access/)). Perhaps it might be interesting to you.

nimobeeren commented:
> @nimobeeren I found this link earlier today ([Claude’s API now supports CORS requests, enabling client-side applications](https://simonwillison.net/2024/Aug/23/anthropic-dangerous-direct-browser-access/)). Perhaps it might be interesting to you.

Thanks, that's good to know! I think this only applies to the API though, rather than the user-facing Claude product. I think there is a place for API-based interactions with LMMs (like Raycast AI), but this script is meant to work with the (free or paid) web interface.
